src/Compilers/CSharp/Portable/FlowAnalysis/PreciseAbstractFlowPass.cs

```diff
@@ -923,30 +923,25 @@ public override BoundNode VisitPassByCopy(BoundPassByCopy node)
         public override BoundNode VisitIsPatternExpression(BoundIsPatternExpression node)
         {
             VisitRvalue(node.Expression);
-            VisitPattern(node.Expression, node.Pattern);
+            VisitPattern(node.Pattern);
+            var reachableLabels = node.DecisionDag.ReachableLabels;
+            if (!reachableLabels.Contains(node.WhenTrueLabel))
+            {
+                SetState(this.StateWhenFalse);
+                SetConditionalState(UnreachableState(), this.State);
```

Comment:
If SetConditionalState zeroes out the current state, what's the point of calling SetState before this?

Reply:
The call to `UnreachableState()` requires as a precondition that the current state is not a split state.

---
In reply to: [180252872](https://github.com/dotnet/roslyn/pull/26035#discussion_r180252872) [](ancestors = 180252872)